lib/oban/queue/executor.ex

```diff
@@ -59,29 +85,28 @@ defmodule Oban.Queue.Executor do
 
   # Helpers
 
-  defp to_module(worker) when is_binary(worker) do
-    worker
-    |> String.split(".")
-    |> Module.safe_concat()
+  defp worker_module(%Job{worker: worker} = job) when is_binary(worker) do
+    {:ok,
+     worker
+     |> String.split(".")
+     |> Module.safe_concat()}
+  rescue
+    exception ->
+      {:failure, job, :exception, exception, __STACKTRACE__}
   end
 
-  defp to_module(worker) when is_atom(worker), do: worker
+  defp worker_module(%Job{worker: worker}) when is_atom(worker), do: {:ok, worker}
 
   # While it is slightly wasteful, we have to convert the worker to a module again outside of
   # `safe_call/1`. There is a possibility that the worker module can't be found at all and we
   # need to fall back to a default implementation.
-  defp worker_backoff(%Job{attempt: attempt, worker: worker}) do
-    module =
-      try do
-        to_module(worker)
-      rescue
-        ArgumentError -> nil
-      end
+  defp worker_backoff(%Job{attempt: attempt} = job) do
+    case worker_module(job) do
+      {:ok, module} ->
+        module.backoff(attempt)
```

Comment:
This is a little riskier than the previous code because we aren't checking that the module exports `backoff/1`. We know the module is real, but that doesn't mean it is actually a worker. We have the same issue with `worker.timeout` now, so perhaps we can verify that it is a worker in `worker_module/1` instead?

Reply:
How about this then:

```
defp worker_backoff(%Job{attempt: attempt} = job) do
    {:ok, module} = worker_module(job)
    module.backoff(attempt)
  rescue
    _ ->
      Worker.default_backoff(attempt)
  end
```

Reply:
Blah, the whole thing is a little fiddly. Here is the issue I have with the version I just posted, which also applies to the version in master too: If I don't include backoff, or I code it badly and it returns an invalid value, it's silently ignored. Let me do a version where we log some useful stuff for users maybe.

Reply:
Let's skip that for now. I can do some refactoring and work on error messaging after this is merged.

Reply:
👍 I'll focus on the core timing functionality then and get a test case in, and then yeah I think a general pass on safely calling callback functions may be worthwhile.